Client/Helpers/FxALoginHelper.swift

```diff
@@ -0,0 +1,210 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Account
+import Deferred
+import Foundation
+import Shared
+import SwiftyJSON
+import XCGLogger
+
+private let applicationDidRequestUserNotificationPermissionPrefKey = "applicationDidRequestUserNotificationPermissionPrefKey"
+
+private let log = Logger.browserLogger
+
+protocol FxAPushLoginDelegate : class {
+    func accountLoginDidFail()
+
+    func accountLoginDidSucceed(withFlags flags: FxALoginFlags)
+}
+
+/// Small struct to keep together the immediately actionable flags that the UI is likely to immediately
+/// following a successful login. This is not supposed to be a long lived object.
+struct FxALoginFlags {
+    let pushEnabled: Bool
+    let verified: Bool
+}
+
+/// This class manages the from successful login for FxAccounts to 
+/// asking the user for notification permissions, registering for 
+/// remote push notifications (APNS), registering for WebPush notifcations
```

Comment:
Nit: Do we support WebPush with this yet? Might be helpful to remove that from the comment to clarify.